packages/sdk/src/credential/onBehalfOfUserCredential.ts

```diff
@@ -72,14 +79,24 @@ export class OnBehalfOfUserCredential implements TokenCredential {
     }
 
     const normalizedAuthorityHost = config.authentication!.authorityHost!.replace(/\/+$/g, "");
-    const authority: string =
-      normalizedAuthorityHost + "/" + config.authentication?.tenantId;
+    const authority: string = normalizedAuthorityHost + "/" + config.authentication?.tenantId;
+    let clientCertificate: ClientCertificate | undefined;
+    if (!config?.authentication?.clientSecret) {
```

Comment:
why we need check clientsecret before parsing certificate?

Reply:
In case, both clientSecret and certificatePath are set. Then no need to parse cert.